helmclienttest/helmclient.go

```diff
@@ -0,0 +1,78 @@
+package helmclienttest
+
+import (
+	"context"
+
+	"github.com/giantswarm/helmclient"
+	"k8s.io/helm/pkg/helm"
+)
+
+type Config struct {
+	DefaultReleaseContent *helmclient.ReleaseContent
+	DefaultReleaseHistory *helmclient.ReleaseHistory
+	DefaultError          error
```

Comment:
I like having a mock in general. But how about changing the approach and make it more flexible? Example below:

```go
type Config struct {
	GetReleaseContentFunc func(ctx context.Context, releaseName string) (*helmclient.ReleaseContent, error)
}
```

And then:

```go
func (c *Client) GetReleaseContent(ctx context.Context, releaseName string) (*helmclient.ReleaseContent, error) {
	if c.getReleaseContentFunc != nil {
		return c.GetReleaseContentFunc()
	}

	return nil, nil
}
```

Such approach would allow to implement anything within the mock w/o having to implement all the methods.

Reply:
Yes that is nice. I don't want to spend lots of time on this but the approach is more flexible. I'll look at it later.

Reply:
Yes, but then we'll have to spend more time on vendoring. You can expose only functions that will be used by you in testing. Extending the object is backward compatible.

Reply:
@kopiczko Looking at this again. Its more flexible but it creates another problem.

https://github.com/giantswarm/chart-operator/blob/1553942b22cbecea7b26b7b90bb56399d3b3d8f6/service/controller/chartconfig/v5/resource/chart/mock_test.go#L50-L56
https://github.com/giantswarm/cluster-operator/blob/04061930d9404bd9b406d68437de3b1c46d31c6f/pkg/v10/chartconfig/mock_test.go#L46-L52

The mocks are the same in multiple projects but by passing in `GetReleaseContentFunc` we need to define the same functions in both. I think its better to extract all the mock logic to this package. WDYT?

Reply:
Ack. I didn't realize this is already in use. Let's leave it as it is.